config.yaml.dist

```diff
@@ -15,24 +15,23 @@
 # This file contains the configuration for installing a single Kubernetes
 # clusters using KubeOne. You can augment some options by providing
 # Terraform output at runtime, like explained in the documentation.
-
+apiVersion: kubeone.io/v1alpha1
+kind: KubeOneCluster
 name: my-demo-cluster
 
 versions:
   kubernetes: "1.14.1"
 
-network:
+clusterNetwork:
   # the subnet used for pods (flannel);
   # leave it empty for default: 10.244.0.0/16
-  pod_subnet: ""
-
+  podSubnet: ""
   # the subnet used for services;
   # leave it empty for default: 10.96.0.0/12
-  service_subnet: ""
-
+  serviceSubnet: ""
   # a nodePort range to reserve for services;
   # leave it empty for default: 30000-32767
-  node_port_range: ""
+  nodePortRange: ""
 
 provider:
```

Comment:
Did you forgot to update to `cloudProvider`? Also, why `provider` is bad?

Reply:
I forgot to update some fields including `cloudProvider`. Pushed a commit that updates everything needed.

The `cloudProvider` change was based on https://github.com/kubermatic/kubeone/pull/353#discussion_r276633959 and some discussion we had on the stand-up meeting.

One of the reasons I found `provider` bad is because it's mostly about the cloud provider features. `provider` wasn't descriptive enough and the idea was to rename it to something more clear.